code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -1749,3 +1749,122 @@ All effects don't start immediately, but rather get worse over time; the rate is
 		M.adjustStaminaLoss(35)
 		. = TRUE
 	..()
+
+/datum/reagent/consumable/ethanol/fruit_wine
+	name = "Fruit Wine"
+	id = "fruit_wine"
+	description = "A wine made from grown plants."
+	color = "#FFFFFF"
+	boozepwr = 35
+	taste_description = "bad coding"
+	var/list/names = list("null fruit" = 1) //Names of the fruits used. Associative list where name is key, value is the percentage of that fruit.
+	var/list/tastes = list("bad coding" = 1) //List of tastes. See above.
+	can_synth = FALSE
+
+/datum/reagent/consumable/ethanol/fruit_wine/on_new(list/data)
+	names = data["names"]
+	tastes = data["tastes"]
+	boozepwr = data["boozepwr"]
+	color = data["color"]
+	generate_data_info(data)
+
+/datum/reagent/consumable/ethanol/fruit_wine/on_merge(list/data, amount)
+	var/diff = (amount/volume)
+	if(diff < 1)
+		color = BlendRGB(color, data["color"], diff/2) //The percentage difference over two, so that they take average if equal.
+	else
+		color = BlendRGB(color, data["color"], (1/diff)/2) //Adjust so it's always blending properly.
+	var/oldvolume = volume-amount
+
+	var/list/cachednames = data["names"]
+	for(var/name in names | cachednames)
+		names[name] = ((names[name] * oldvolume) + (cachednames[name] * amount)) / volume
+
+	var/list/cachedtastes = data["tastes"]
+	for(var/taste in tastes | cachedtastes)
+		tastes[taste] = ((tastes[taste] * oldvolume) + (cachedtastes[taste] * amount)) / volume
+
+	boozepwr *= oldvolume
+	var/newzepwr = data["boozepwr"] * amount
+	boozepwr += newzepwr
+	boozepwr /= volume //Blending boozepwr to volume.
+	generate_data_info(data)
+
+/datum/reagent/consumable/ethanol/fruit_wine/proc/generate_data_info(list/data)
+	var/minimum_percent = 0.15 //Percentages measured between 0 and 1.
+	var/list/primary_tastes = list()
+	var/list/secondary_tastes = list()
+	glass_name = "glass of [name]"
+	glass_desc = description
+	for(var/taste in tastes)
+		if(tastes[taste] < minimum_percent)
+			continue
+		if(tastes[taste] >= minimum_percent*2)
+			primary_tastes += taste
+		else if(tastes[taste] >= minimum_percent)
+			secondary_tastes += taste
+
+	var/minimum_name_percent = 0.35
+	name = ""
+	var/highest = 0
+	var/highest_name = ""
+	var/named = FALSE
+	for(var/fruit_name in names)
```

Comment:
You can do this easier by sorting the list with `sortTim(names, /proc/cmp_numeric_dsc, TRUE)` and then grab names[1] for the highest name and names[names[1]] for the value

Reply:
I had no idea this existed, thank you.

Reply:
look at code/__HELPERS/cmp.dm if you want to implement your own comparison function